File: app/src/main/java/com/me/myapplication/activity/DailyTestActivity.java
package com.me.myapplication.activity;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.viewpager.widget.ViewPager;

import com.google.gson.Gson;
import com.me.myapplication.activity.adapter.CardItemDetail;
import com.me.myapplication.activity.adapter.CardPagerAdapterDetail;
import com.me.myapplication.activity.model.ShadowTransformer;
import com.me.myapplication.R;

//import me.zqn.astrologyapp.bean.DailyAstro;
//import me.zqn.astrologyapp.util.OkHttpUtils;
//import me.zqn.astrologyapp.util.UrlUtil;

//public class ViewPaperActivity extends AppCompatActivity implements View.OnClickListener,
//        CompoundButton.OnCheckedChangeListener {

//public class ViewPaperActivity extends AppCompatActivity implements CompoundButton.OnCheckedChangeListener {
//public class DailyTestActivity extends AppCompatActivity implements OkHttpUtils.callBack{
public class DailyTestActivity extends AppCompatActivity {
//    private Button mButton;
    private ViewPager mViewPager;

    private CardPagerAdapterDetail mCardAdapter;
    private ShadowTransformer mCardShadowTransformer;
    private String astro;
    private String url1;
//    private String url2;
//    private String url3;
    private Gson gson;
    private String str1;
    private String str2;
    private String str3;

//    private CardFragmentPagerAdapter mFragmentCardAdapter;
//    private ShadowTransformer mFragmentCardShadowTransformer;

//    private boolean mShowingFragments = false;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_view_paper);

        astro = getIntent().getExtras().getString("astro");
        mViewPager = (ViewPager) findViewById(R.id.viewPager);
//        mButton = (Button) findViewById(R.id.search_button);
//        mButton = (Button) findViewById(R.id.cardTypeBtn);
//        ((CheckBox) findViewById(R.id.checkBox)).setOnCheckedChangeListener(this);
//        mButton.setOnClickListener(this);

//        url1 = UrlUtil.start1.replace("%d",astro);
//        gson = new Gson();
//        OkHttpUtils.callBackUIDataFormatOne(url1,OkHttpUtils.TYPE_TEXT,this);

        mCardAdapter = new CardPagerAdapterDetail();
        mCardAdapter.addCardItem(new CardItemDetail(R.string.title_1, astro));
        mCardAdapter.addCardItem(new CardItemDetail(R.string.title_2, str2));
        mCardAdapter.addCardItem(new CardItemDetail(R.string.title_3, str3));

////        mCardAdapter.addCardItem(new CardItem(R.string.title_4, R.string.text_1));
//        mFragmentCardAdapter = new CardFragmentPagerAdapter(getSupportFragmentManager(),
//                dpToPixels(2, this));

        mCardShadowTransformer = new ShadowTransformer(mViewPager, mCardAdapter);
//        mFragmentCardShadowTransformer = new ShadowTransformer(mViewPager, mFragmentCardAdapter);

        mViewPager.setAdapter(mCardAdapter);
        mViewPager.setPageTransformer(false, mCardShadowTransformer);
        mViewPager.setOffscreenPageLimit(3);
    }

//    @Override
//    public void onClick(View view) {
//
//    }

//    public static float dpToPixels(int dp, Context context) {
//        return dp * (context.getResources().getDisplayMetrics().density);
//    }
//
//    @Override
//    public void callBackUIString(String data) {
//         DailyAstro astroBean = gson.fromJson(data, DailyAstro.class);
//         str1 = astroBean.getSummary();
//         str2 = astroBean.getWork() + '\n' + astroBean.getMoney();
//         str3 = astroBean.getSummary();
//    }
//
//    @Override
//    public void callBackUIByte(byte[] datas) {
//
//    }

//    @Override
//    public void onCheckedChanged(CompoundButton compoundButton, boolean b) {
//        mCardShadowTransformer.enableScaling(b);
////        mFragmentCardShadowTransformer.enableScaling(b);
//    }




}

File: app/src/main/java/com/me/myapplication/activity/DailyTestActivity.kt
package com.me.myapplication.activity

import android.content.Context
import android.os.Bundle
import android.os.Handler
import android.os.TestLooperManager

import android.view.View
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.viewpager.widget.ViewPager

import com.google.gson.GsonBuilder
import com.me.myapplication.R
import com.me.myapplication.activity.adapter.CardPagerAdapterDetail
import com.me.myapplication.activity.bean.DailyAstro
import com.me.myapplication.activity.interceptor.TimeConsumeInterceptor
import com.me.myapplication.activity.model.CardItemDetail
import com.me.myapplication.activity.model.ShadowTransformer
import okhttp3.*
import java.io.IOException
import java.util.concurrent.TimeUnit


class DailyTestActivity : AppCompatActivity() {

    private var mViewPager: ViewPager? = null
    private var mCardAdapter: CardPagerAdapterDetail? = null
    private var mCardShadowTransformer: ShadowTransformer? = null
    private var astro: String? = null
    private var textView: TextView? = null

    //    private String url2;
    //    private String url3;

    var str1: String? = null
    var str2: String? = null
    var str3: String? = null

    var mcontext: Context? = null
    var cache: Cache? = null

    val okhttpListener = object : EventListener() {}

    val gson = GsonBuilder().create()
    var client: OkHttpClient? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
            setContentView(R.layout.activity_daily_test)
            textView = findViewById(R.id.today_info)
//            str1 = "信息每天都有更新~\n每天都记得回来看看哦~"
//            str2 = "如果你向神求助，说明你相信神的能力。如果神没有帮助你，说明神相信你的能力。"
            str1 = "今天会开心~"
            str2 = "每天都是100分~"

            astro = intent.extras!!.getString("astro")

            mcontext = applicationContext

            client = OkHttpClient
                .Builder()
                .cache(cache)
                .addInterceptor(TimeConsumeInterceptor())
                .eventListener(okhttpListener)
                .build()

            click()

            mViewPager = findViewById<View>(R.id.viewPager) as ViewPager
            mCardAdapter = CardPagerAdapterDetail()
            mCardAdapter!!.addCardItem(CardItemDetail(R.string.title_1, str1))
            mCardAdapter!!.addCardItem(CardItemDetail(R.string.title_2, str2))
//            mCardAdapter!!.addCardItem(CardItemDetail(R.string.title_3, str3))
            mCardShadowTransformer = ShadowTransformer(mViewPager, mCardAdapter)
            mViewPager!!.adapter = mCardAdapter
            mViewPager!!.setPageTransformer(false, mCardShadowTransformer)
            mViewPager!!.offscreenPageLimit = 3
        }

        fun request(url: String, callback: Callback) {
            val request: Request = Request.Builder()
                .cacheControl(CacheControl.Builder().maxStale(10, TimeUnit.DAYS).build())
                .url(url)
                .build()
            client?.newCall(request)?.enqueue(callback)
        }

        fun click() {
            val url =
                "http://web.juhe.cn/constellation/getAll?consName=${astro}&type=today&key=994c7f89c7d8012fe43ae2cd2d736302"

            request(url, object : Callback {


                override fun onFailure(call: Call, e: IOException) {
                    str1 = e.message
                }

                override fun onResponse(call: Call, response: Response) {

                    val bodyString = response.body?.string()
                    val dailyAstro = gson.fromJson(bodyString, DailyAstro::class.java)
                    runOnUiThread {
                        if (dailyAstro.all == null) {
                            textView?.text = "您的星星可能走丢了哦~"
                        } else {
                            textView?.text = "${astro}的您今天运势如下:\n\n" +
                                    "${dailyAstro.summary}\n" +
                                    "今天总分是${dailyAstro.all}"
                        }
                    }
                }
            })
        }
    }
